torch_geometric/data/materialized_graph.py

```diff
@@ -0,0 +1,125 @@
+from abc import abstractmethod
+from collections.abc import MutableMapping
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, Dict, Optional, Union
+
+from torch import LongTensor
+
+from torch_geometric.typing import EdgeTensorType, EdgeType
+from torch_geometric.utils.mixin import CastMixin
+
+
+class EdgeLayout(Enum):
+    COO = 'COO'
+    CSC = 'CSC'
+    CSR = 'CSR'
+    LIL = 'LIL'
+
+
+@dataclass
+class EdgeAttr(CastMixin):
```

Comment:
This term is now a bit overloaded.

Reply:
In what manner? I think it aligns well with `TensorAttr` in `FeatureStore` (they are both attributes of objects we are trying to add to the respective stores).

Reply:
Ah, right. I think it's probably okay to leave this overloaded term for now (mostly since I can't think of a better one), and if you have any ideas we can do a quick refactor.